jib-core/src/main/java/com/google/cloud/tools/jib/http/FailoverHttpClient.java

```diff
@@ -110,6 +120,8 @@ private static HttpTransport getInsecureHttpTransport() {
   private final Supplier<HttpTransport> secureHttpTransportFactory;
   private final Supplier<HttpTransport> insecureHttpTransportFactory;
 
+  private final ConcurrentHashMap<String, Failover> failoverHistory = new ConcurrentHashMap<>();
```

Comment:
This makes me wonder if we should add any tests for concurrency. Maybe not worth the trouble.

Reply:
Good point. But I can't think of a practical way to testing concurrency in a deterministic way (or what really to verify thereof), particularly in this case.

Reply:
Yeah, I think using a thread-safe data structure is enough for the changes in this PR. Although maybe it's something to think about in the future.